src/daemon/jsonrpc/api.rs

```diff
@@ -1075,9 +1076,49 @@ impl RpcApi for RpcImpl {
                     continue;
                 }
             }
+
+            let spent_vaults = db_vaults_from_spend(&db_path, &db_spend.psbt.txid())
+                .map_err(|e| internal_error!(e))?;
+            let derivation_index = spent_vaults
+                .values()
+                .map(|v| v.derivation_index)
+                .max()
+                .expect("Spent vaults should not be empty");
```

Comment:
Did you check that assumption? It seems to me that `updatespendtx` would happily insert a Spend refering to 0 vaults in the DB!
https://github.com/revault/revaultd/blob/062abff12e2dc3800b2bdf21b8853adae6a6df92/src/daemon/jsonrpc/api.rs#L1001-L1031

So i agree we should keep this assumption but we should open an issue for the `updatespendtx` "bug" then, good catch :)

Reply:
Actually since the function parameter is typed, we would never accept a Spend without input (it would fail to be parsed). And all spend inputs must be unvaults, so we good.

Leaving that here still as this could use some documentation! (not your unwrap, the updatespendtx code)

Reply:
> we would never accept a Spend without input (it would fail to be parsed)
I don't understand. `updatespendtx` takes a `SpendTransaction`, which doesn't necessarily have the inputs set: https://github.com/revault/revault_tx/blob/master/src/transactions.rs#L1178

Reply:
This is what is written in the comment **just above** yours..

Reply:
https://github.com/revault/revault_tx/blob/6a0c2759332e4674e7a1f288241b0a546422d4f5/src/transactions.rs#L1391-L1393